samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,401 @@
+            resource_type = resource_dict.get("Type")
+
+            if resource_type in CODE_SYNCABLE_RESOURCES:
+                last_resource_dict = last_deployed_template_dict.get("Resources", {}).get(resource_logical_id, {})
+                resource_resolved_id = nested_prefix + resource_logical_id if nested_prefix else resource_logical_id
+
+                if resource_type == AWS_LAMBDA_FUNCTION:
+                    if not resource_dict.get("Properties", {}).get("Code", None) == last_resource_dict.get(
+                        "Properties", {}
+                    ).get("Code", None):
+                        self._code_sync_resources.add(ResourceIdentifier(resource_resolved_id))
+                else:
+                    for field in GENERAL_REMOVAL_MAP.get(resource_type, []):
+                        if not resource_dict.get("Properties", {}).get(field, None) == last_resource_dict.get(
+                            "Properties", {}
+                        ).get(field, None):
+                            self._code_sync_resources.add(ResourceIdentifier(resource_resolved_id))
+
+            if resource_type in SYNCABLE_STACK_RESOURCES:
+                try:
+                    stack_resource_detail = self._cfn_client.describe_stack_resource(
+                        StackName=stack_name, LogicalResourceId=resource_logical_id
+                    )
+                except ClientError as ex:
+                    LOG.debug(
+                        "Cannot get resource detail with name %s on CloudFormation", resource_logical_id, exc_info=ex
+                    )
+                    return False
+
+                # If the nested stack is of type AWS::CloudFormation::Stack,
+                # The template location will be under TemplateURL property
+                # If the nested stack is of type AWS::Serverless::Application,
+                # the template location will be under Location property
+                template_field = "TemplateURL" if resource_type == AWS_CLOUDFORMATION_STACK else "Location"
+                template_location = resource_dict.get("Properties", {}).get(template_field)
+
+                # For AWS::Serverless::Application, location can be a ApplicationLocationObject dict containing SAR ID
+                if isinstance(template_location, dict):
+                    continue
+                # For other scenarios, template location will be a string (local or s3 URL)
+                elif not self._auto_skip_infra_sync(
+                    resource_dict.get("Properties", {}).get(template_field),
+                    current_built_template.get("Resources", {})
+                    .get(resource_logical_id, {})
+                    .get("Properties", {})
+                    .get(template_field),
+                    stack_resource_detail.get("StackResourceDetail", {}).get("PhysicalResourceId", ""),
+                    resource_logical_id + "/",
```

Comment:
`nested_prefix` should be included here as well for stacks with more than 2 levels of nesting.